src/main/java/com/shapesecurity/shift/parser/Parser.java

```diff
@@ -27,9 +27,13 @@
 import org.jetbrains.annotations.Nullable;
 
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.function.BiFunction;
 
 public abstract class Parser extends Tokenizer {
+    public Map<Node, Maybe<SourceSpan>> locations = new HashMap<>();
```

Comment:
We prefer to use the [shape-functional-java library](https://github.com/shapesecurity/shape-functional-java) (com.shapesecurity.functional.*) for these kinds of data structures. Please use the `HashTable<K, V>` from there. Also, you won't need to parameterise it by a `Maybe<SourceSpan>`, just `SourceSpan` because `get` returns a `Maybe<V>`.